src/lib/mongoBackend/mongoSubCache.cpp

```diff
@@ -13,10 +40,237 @@ using std::auto_ptr;
 
 /* ****************************************************************************
 *
+* EntityInfo - 
+*
+* The struct fields:
+* -------------------------------------------------------------------------------
+* o entityIdPattern      regex describing EntityId::id (OMA NGSI type)
+* o entityType           string containing the type of the EntityId
+*
+*/
+typedef struct EntityInfo
+{
+  regex_t       entityIdPattern;
+  std::string   entityType;
+  bool          entityIdPatternToBeFreed;
+
+  EntityInfo() {}
+  EntityInfo(const std::string& idPattern, const std::string& _entityType);
+  ~EntityInfo() { release(); }
+
+  bool          match(const std::string& idPattern, const std::string& type);
+  void          release(void);
+  void          present(const std::string& prefix);
+} EntityInfo;
+
+
+
+/* ****************************************************************************
+*
+* EntityInfo::EntityInfo - 
+*/
+EntityInfo::EntityInfo(const std::string& idPattern, const std::string& _entityType)
+{
+  regcomp(&entityIdPattern, idPattern.c_str(), 0);
+
+  entityType               = _entityType;
+  entityIdPatternToBeFreed = true;
+}
+
+
+
+/* ****************************************************************************
+*
+* EntityInfo::match - 
+*/
+bool EntityInfo::match
+(
+  const std::string&  id,
+  const std::string&  type
+)
+{
+  //
+  // If type non-empty - perfect match is mandatory
+  // If type is empty - always matches
+  //
+  if ((type != "") && (entityType != type) && (entityType != ""))
+  {
+    return false;
+  }
+
+  // REGEX-comparison this->entityIdPattern VS id
+  return regexec(&entityIdPattern, id.c_str(), 0, NULL, 0) == 0;
+}
+
+
+
+/* ****************************************************************************
+*
+* EntityInfo::release - 
+*/
+void EntityInfo::release(void)
+{
+  if (entityIdPatternToBeFreed)
+  {
+    regfree(&entityIdPattern);
+    entityIdPatternToBeFreed = false;
+  }
+}
+
+
+
+/* ****************************************************************************
+*
+* EntityInfo::present - 
+*/
+void EntityInfo::present(const std::string& prefix)
+{
+  LM_F(("%sid:    regex describing EntityId::id (the original string is not saved)", prefix.c_str()));
```

Comment:
Maybe LM_M would me this more easy to "catch" to be removed at the end? (I think that LM_F are used for present() methods, which is ok, but probably I don't remembering well...).